src/__tests__/test-utils-selectors.test.tsx

```diff
@@ -0,0 +1,144 @@
+      } else if (element.type === 'TemplateElement') {
+        literal += element.value.raw;
+      } else if (element.type === 'Identifier') {
+        literal += '*';
+      } else if (element.type === 'NumericLiteral') {
+        literal += element.value;
+      } else {
+        throw new Error(`Unhandled template literal type at ${file}:${node.loc?.start.line}.`);
+      }
+    }
+    return literal;
+  }
+
+  function extractor(): PluginObj {
+    const selectorToFilePath = new Map<string, string>();
+    return {
+      visitor: {
+        // Find import statements for selectors.
+        ImportDeclaration(path: NodePath<types.ImportDeclaration>) {
+          if (path.node.source.value.endsWith('selectors.js') && path.node.specifiers.length === 1) {
+            const specifier = path.node.specifiers[0];
+            if (specifier.type === 'ImportDefaultSpecifier') {
+              selectorToFilePath.set(specifier.local.name, resolveSelectorsPath(path.node.source.value));
+            } else {
+              throw new Error(`Unhandled styles import type at ${file}:${path.node.loc?.start.line}.`);
+            }
+          }
+        },
+        // Find selector references and extract used property names.
+        MemberExpression(path: NodePath<types.MemberExpression>) {
+          if (path.node.object.type === 'Identifier' && selectorToFilePath.has(path.node.object.name)) {
+            onExtract(selectorToFilePath.get(path.node.object.name)!, getPropertyName(path));
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [...defaultPlugins, extractor] })?.code;
+}
+
+function extractComponentSelectors(file: string, usedProperties: string[], onExtract: (selector: string) => void) {
+  function extractor(): PluginObj {
+    return {
+      visitor: {
+        ObjectProperty(path: NodePath<types.ObjectProperty>) {
+          if (path.node.key.type === 'StringLiteral' && matchProperties(usedProperties, path.node.key.value)) {
+            if (path.node.value.type === 'StringLiteral') {
+              onExtract(trimSelectorHash(path.node.value.value));
+            } else {
+              throw new Error(`Unhandled selector value type at ${file}:${path.node.loc?.start.line}.`);
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [...defaultPlugins, extractor] })?.code;
+}
+
+function trimSelectorHash(selector: string) {
+  const splitSelector = selector.replace('.', '').split('_');
+  if (splitSelector.length >= 5) {
+    splitSelector.splice(splitSelector.length - 2, splitSelector.length);
+    return splitSelector.join('_');
+  }
+  return selector;
+}
+
+function resolveSelectorsPath(importPath: string) {
+  return 'lib/components/' + importPath.replace(/\.\.\//g, '');
```

Comment:
Still unclear what is happening here. Why did you not use `path.resolve(from, to)` to follow relative paths?

Reply:
The import path is taken from `src/test-utils/selectors/{component}`. The selectors path should resolve to `lib/components/{component}/styles.selectors.js`.

If using `Path.resolve` this can be achieved with e.g.
```
Path.resolve(testUtilFilePath, importPath).replace('/src/', '/lib/')
```

I like the current approach better.

Though, this needs to be parametrised somewhat when moving the extractor part to the test-utils package so that it can be reused.

Reply:
The import path is taken from `src/test-utils/selectors/{component}`. The selectors path should resolve to `lib/components/{component}/styles.selectors.js`.

If using `Path.resolve` this can be achieved with e.g.
```
Path.resolve(testUtilFilePath, importPath).replace('/src/', '/lib/')
```

I like the current approach better.

Though, this needs to be parametrised somewhat when moving the extractor part to the test-utils package so that it can be reused.

Reply:
The solution does support nested paths like `../../../app-layout/test-classes/styles.selectors.js` but I will give your solution a try

Reply:
Updated